docs/protocol_overview.md

```diff
@@ -0,0 +1,245 @@
+3. **Communication Channel**: a mechanism to communicate with each other in a decentralized, verifiable, and fault-tolerant way.
+
+## Basics and Terminologies
+
+The reader of this document should be familiar with the basic concepts of blockchain.
+
+- Simperby is an engine that can build an instance of a blockchain.
+  Any blockchain built with Simperby is sometimes called a *Simperby chain*.
+- The unit of state transition is called a *transaction*.
+  `StoreData`, `DelegateGovernanceVotingPower`, `AddValidator` are the examples.
+- A sequence of transactions is called an *agenda*.
+  **Any governance voting is performed on an agenda and the target block height**.
+- A *block* consists of the header and **a single agenda, which is mandatory**.
+- The participants of the governance are called *members*.
+- The participants of the consensus are called *validators*, which is a subset of the *members*.
+- A *consensus leader* or a *block proposer* is a validator that proposes a block for the round.
+- A *round* is a period of time in which a consensus leader proposes a block, and consensus votes(pre-vote, pre-commit) are performed. Refer to Tendermint for more details.
+- A validator is *honest* or *not byzantine* if they follow the consensus protocol.
+- A block proposer is *faithful* if they fulfill their responsibility and *good* if they don't overuse their power. They are *lazy* and *bad* otherwise.
+  The notion of *responsibility* and *power* will be explained [later](#consensus-leader).
+
+## Summary
+
+Here is a summarized version of the Simperby protocol.
+
+1. We want the node operation to be **simple and lightweight** to realize a truly distributed, decentralized and self-hosted organization.
+2. To be so, it is assumed that most of the nodes are **rarely online** and the protocol produces new blocks **on-demand**.
+3. To accomplish that, the **consensus round must be very long**.
+4. For every block, members can **vote on or propose an agenda, propagating** to each other.
+5. At the same time, members can also **propagate their chats** used for the communication, which is **coordinated by the block proposer**.
+6. If there is a **governance-approved (majority-voted) agenda** on the network, the block proposer should **include it in the block along with the chat log**, and propose to the consensus.
+7. In that the block proposer should be online most of the time (chat coordination and block proposing), this **responsibility should be laid on a few of the validators most of the time**, to ensure the rarely-online assumption.
+8. Also the role of **block proposer has some authorities** (chat coordination and agenda inclusion) that can't be cryptographically verified if misused (typically censorship).
+9. Thus there exists **'few of the validators' that can be lazy or commit harmful misuses** of their authorities which aren't really byzantine faults or invalid blocks.
```

Comment:
```suggestion
9. Thus there exists **'few validators' that can be lazy or commit harmful misuses** of their authorities which aren't really byzantine faults like proposing invalid blocks.
```

Reply:
The meaning has beeen changed.
Mine: there is a privileged set of validators that must be moderated.
Yours: it is natural that some validators may become be lazy or harmful.

Reply:
Also, I distinguished 'byzantin fault' and 'invalid block' but you included the latter in the former.

Reply:
> Also, I distinguished 'byzantin fault' and 'invalid block' but you included the latter in the former.

ok I understand

Reply:
> Mine: there is a privileged set of validators that must be moderated.

Well, I'm not sure the readers will comprehend what you mean with the original sentence.
I'll resolve this conversation if other reviewers are happy with this.

Reply:
I changed it to a super explicit expression.

Reply:
great